<launch>

  <arg name="rpmRate" default="10.0"/>
  <arg name="minRpmRate" default="5.0"/>
  <arg name="maxRpmRate" default="20.0"/>
  <arg name="tempRate" default="2.0"/>
  <arg name="minTempRate" default="1.0"/>
  <arg name="maxTempRate" default="4.0"/>
  <arg name="motorTemperatureSteadyBand" default="0.0"/>
  <arg name="setRPMTimeout" default="0.5"/>
  <arg name="canNodeId1" default="0x07"/>
  <arg name="canNodeId2" default="0x08"/>
  <arg name="motorTemperatureThreshold" default="50.0"/>
  <arg name="maxAllowedMotorRPM" default="0.0"/>
  <arg name="currentLoggingEnabled" default="false"/>
  <arg name="reportBatteryHealthRate" default="1.0"/>
  <arg name="minReportBatteryHealthRate" default="0.5"/>
  <arg name="maxReportBatteryHealthRate" default="2.0"/>
  <arg name="maxBatteryTotalCurrent" default="1"/>
  <arg name="minBatteryCellVoltage" default="1"/>
  <arg name="batteryTotalCurrentSteadyBand" default="0.0"/>

  <node name="thruster_control_node" pkg="thruster_control" type="thruster_control_node" output="screen">
    <param name="report_rpm_rate" type="double" value="$(arg rpmRate)" />
    <param name="min_report_rpm_rate" type="double" value="$(arg minRpmRate)" />
    <param name="max_report_rpm_rate" type="double" value="$(arg maxRpmRate)" />
    <param name="report_motor_temperature_rate" type="double" value="$(arg tempRate)" />
    <param name="min_report_motor_temperature_rate" type="double" value="$(arg minTempRate)" />
    <param name="max_report_motor_temperature_rate" type="double" value="$(arg maxTempRate)" />
    <param name="motor_temperature_steady_band" type="double" value="$(arg motorTemperatureSteadyBand)"/>
    <param name="set_rpm_timeout_seconds" type="double" value="$(arg setRPMTimeout)" />
    <param name="current_logging_enabled" type="boolean" value="$(arg currentLoggingEnabled)" />
    <param name="can_node_id_1" type="string" value="$(arg canNodeId1)" />
    <param name="can_node_id_2" type="string" value="$(arg canNodeId2)" />
    <param name="motor_temperature_threshold" type="double" value="$(arg motorTemperatureThreshold)" />
    <param name="max_allowed_motor_rpm" type="double" value="$(arg maxAllowedMotorRPM)" />
    <param name="report_battery_health_rate" type="double" value="$(arg reportBatteryHealthRate)" />
    <param name="battery_total_current_steady_band" type="double" value="$(arg batteryTotalCurrentSteadyBand)"/>
    <param name="min_report_battery_health_rate" type="double" value="$(arg minReportBatteryHealthRate)" />
    <param name="max_report_battery_health_rate" type="double" value="$(arg maxReportBatteryHealthRate)" />
    <param name="max_battery_total_current" type="double" value="$(arg maxBatteryTotalCurrent)"/>
    <param name="min_battery_cell_voltage" type="double" value="$(arg minBatteryCellVoltage)"/>
  </node>

</launch>
